test/abstract_syntax_tree.coffee

```diff
@@ -2727,6 +2731,21 @@ test "AST as expected for While node", ->
     postfix: no
     loop: yes
 
+  testExpression '''
+    x = (z() while y)
+  ''',
+    type: 'AssignmentExpression'
+    right:
+      type: 'WhileStatement'
+      body:
+        type: 'BlockStatement'
+        body: [
+          type: 'ExpressionStatement'
+          expression:
+            type: 'CallExpression'
+            returns: yes
```

Comment:
The "return value" of accumulated `while`/`for` loops should be marked as `returns: yes`